File: stock/src/main/java/com/vjcspy/spring/packages/stockinfo/service/CorEntityService.java
/* (C) 2024 */
package com.vjcspy.spring.packages.stockinfo.service;

import com.vjcspy.spring.packages.stockinfo.dto.CorEntityDto;
import com.vjcspy.spring.packages.stockinfo.entity.CorEntity;
import com.vjcspy.spring.packages.stockinfo.mapper.CorEntityMapper;
import com.vjcspy.spring.packages.stockinfo.repository.CorEntityRepository;
import java.util.List;
import java.util.stream.Collectors;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class CorEntityService {

    private final CorEntityRepository corEntityRepository;

    @Autowired
    public CorEntityService(CorEntityRepository corEntityRepository) {
        this.corEntityRepository = corEntityRepository;
    }

    /**
     * Lấy danh sách tất cả CorEntity và chuyển sang CorEntityDto
     *
     * @return List<CorEntityDto>
     */
    public List<CorEntityDto> getAllCorEntities() {
        List<CorEntity> entities = corEntityRepository.findAll();
        return entities.stream().map(CorEntityMapper.INSTANCE::toDto).collect(Collectors.toList());
    }
}

File: projects/stock/src/main/kotlin/com/vjcspy/spring/packages/stockinfo/service/CorEntityService.kt
// (mr.vjcspy@gmail.com) 2024
package com.vjcspy.spring.packages.stockinfo.service

import com.vjcspy.kotlinutilities.log.KtLogging
import com.vjcspy.spring.packages.stockinfo.dto.CorEntityDto
import com.vjcspy.spring.packages.stockinfo.entity.CorEntity
import com.vjcspy.spring.packages.stockinfo.mapper.CorMapper
import com.vjcspy.spring.packages.stockinfo.repository.CorEntityRepository
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
class CorEntityService(
    private val corEntityRepository: CorEntityRepository,
) {
    private val logger = KtLogging.logger()

    /**
     * Lấy danh sách tất cả CorEntity và chuyển sang CorEntityDto
     *
     * @return List<CorEntityDto>
     */
    fun getAllCorEntities(): List<CorEntityDto> {
        val entities = corEntityRepository.findAll()
        return entities.map(CorMapper.INSTANCE::toDto)
    }

    @Transactional
    fun saveCorEntity(corEntity: CorEntity): CorEntity = corEntityRepository.save(corEntity)

    @Transactional
    fun saveCorEntities(corEntities: List<CorEntity>): List<CorEntity> {
        logger.info("Saving cor entities")
        return corEntityRepository.saveAll(corEntities)
    }

    @Transactional
    fun deleteAllCorEntities() {
        logger.info("Deleting all cor entities")
        return corEntityRepository.deleteAll()
    }
}
